system-addon/lib/HighlightsFeed.jsm

```diff
@@ -77,6 +83,7 @@ this.HighlightsFeed = class HighlightsFeed {
         break;
       case at.PLACES_BOOKMARK_ADDED:
       case at.PLACES_BOOKMARK_REMOVED:
+      case at.TOP_SITES_UPDATED:
         this.fetchHighlights(false);
```

Comment:
Hrmm... if we're seeing a TOP_SITES_UPDATED, it will probably cause content to show the new top sites, so then we should maybe broadcast too? Otherwise without broadcast, the highlights will still show some things that should have been deduped.. maybe?

Reply:
Yeah, I went back and forth on it. I guess we can broadcast because highlights are below the fold anyway so it wont move things around for the user?

Reply:
Oh. And also, TopSitesFeed is watching for a lot of these same actions to then refresh. Seems like we might double fetchHighlights

Reply:
So if we don't broadcast, we're okay.. ?

Reply:
The way I'm testing right now is bookmarking a default top site. On first run, it shows the bookmark and next new tab has it removed.

Reply:
well hmm. So if the user starts editing their top sites, we will potentially move the highlights around for them while they do that. That seems annoying, but it's probably not visible to them unless they dont have the Top Stories section. So I guess I could go either way.